src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -199,6 +213,24 @@ private List<String> getAllProtectedSystemIndices(final Resolved requestedResolv
         return new ArrayList<>(superAdminAccessOnlyIndexMatcher.getMatchAny(requestedResolved.getAllIndices(), Collectors.toList()));
     }
 
+    /**
+     * Checks whether the request contains any of the indices reserved by extension
+     * @param requestedResolved request to be checked for
+     * @param reservedIndices the list of indices reserved by extension
+     * @return true if it contains reserved index or if system indices are disabled,
+     * false if request doesn't contain any reserved index
+     */
+    private boolean matchAllReservedIndices(final Resolved requestedResolved, final Set<String> reservedIndices) {
```

Comment:
This is meant for the service account logic I believe. I could be mistaken but if this is for the service account logic, we can probably make this more straightforward--just check whether the index being targeted with a request is the one which matches the system index associated with the extension.